### DISCUSSION TITLE
How can this be achieved? Site updates and versioning

### DISCUSSION THREAD

**Discussion by hansfpc** [react-hook-form/react-hook-form#4675] [Q&A] [answered]
How can this be achieved? Site updates and versioning

I want to implement this on my site 🙏🏻

![image](https://user-images.githubusercontent.com/12705403/113914814-f759ee80-97ab-11eb-9fcc-ec723c21d11d.png)

**Comment by jorisre** [comment, marked as answer]
Hi @hansfpc 

The documentation website is made with Gatsby, and use [`onServiceWorkerUpdateReady` Gatsby browser api](https://www.gatsbyjs.com/docs/reference/config-files/gatsby-browser/#onServiceWorkerUpdateReady) to show the message

The code: https://github.com/react-hook-form/documentation/blob/master/gatsby-browser.js#L17

All the website settings are persisted in the sessionStorage by using the [Little State Machine](https://github.com/bluebill1049/little-state-machine) library.

The implementation: https://github.com/react-hook-form/documentation/blob/master/gatsby-browser.js#L7
